src/api/procedures/consumeJoinIdentityAuthorization.ts

```diff
@@ -25,6 +26,8 @@ export interface Storage {
 
 /**
  * @hidden
+ *
+ * Consumes JoinIdentity and RotatePrimaryKeyToSecondaryKey Authorizations
```

Comment:
Initially I made a duplicate procedure, but Sonar Cloud didn't like that. I like the consistency of having the name line up with the auth so its easy to fine. That said I think the duplicate line check is right in that its a fair amount of extra code.

I was thinking of updating this name, but in some sense the rotate auth is a join identity auth, except the added key become primary.

I also was thinking of putting the insides in a shared function, but since the consumes would be siblings in a rather big directory there isn't an obvious place for it. Not exactly a util function.

For now I am thinking either share the function like this makes the most sense.

Reply:
Sharing sounds good to me but only thing is that can we change the name that reflects both the actions something like `AddOrRotateAccount` (wherever JoinIdentity is referred here). 
This is the best I could relate to (`JoinIdentity` eventually adds an account, considering we are referring everything as account and not key, and `RotatePrimaryKeyToSecondaryKey` shortened to `RotateAccount` IDK but changing the name would be better.

Reply:
About the auth names, the "Key" part got left in so it matches with whats on chain. We could rename the variables to something else, but keep the values the same.

I think `AddOrRotateAccount` makes sense as a name for this.

Reply:
Went with `JoinOrRotate`. Almost the same, but I like the `Join` as a bit of a hint.